### PR TITLE
Adds the Game Master and Bionic Game Master professions

### PR DESCRIPTION
**Pull request by ghost**
Popped into my head on Discord, so I made the most of it.

The default Game Master starts with casual wear, a messenger bag, a box of pizza, a can of cola, and a Dungeon Master's Guide, which is a new book that can raise speech to 5. They also have the speech skill at 2 and the survival skill at 1. It costs 2 points to choose.

The Bionic Game Master has memorized the entire DMG, and so doesn't have one. Instead of pizza and cola, they start with caffeine pills. In addition to their bionic setup (Battery System, Cerebral Booster, Enhanced Memory Banks, and one Power Storage), they have a speech skill of 3, melee and survival skills of 2, and all of their melee weapon skills, plus archery and throwing, start at 1. It costs 4 points to choose.

Both professions also start with a unique trait called Dungeon Master, which represents their knowledge of speed-reading and picking out what they need to learn from a volume full of fluff. The trait increases reading speed by a modest 10% (as opposed to Fast Reader's 20%), and, in addition, removes the penalty for reading books beyond your Intelligence requirement. With the trait, as long as you have the skill requirements to read a book, you can do so at full speed! This is to distinguish it from other professions and make it more interesting to run.

### CODE REVIEW

**Comment by Inglonias** [comment]
Interesting. Odd. But definitely interesting. I like the ability to see the hp left on enemies, but wonder if it would make sense to lock that away behind this profession and ONLY this profession. Since it's technically a mutation, we can access it through the debug menu, right? I'm wondering if making it an optional advantage for all characters is a good idea or not.

**Comment by Xhuis** [comment]
It's debug-accessible, yeah. I put it here since I felt like it'd be thematically interesting and serve to make the profession differ in ways other than just starting equipment, but I'm open to other suggestions.

**Comment by nexusmrsep** [comment]
What qualities or skills or I dunno... insight does a game master have to gain knowledge about zombies health? I'd rather believe that somone competent in first aid could quickly judge the extent of wounds inflicted to a creature, but you don't become medical god from playing some DnD. This idea sounds too magical to me to be in a core game, but in a mod - maybe.

**Comment by Xhuis** [comment]
It's a fair point, but I wasn't quite sure what else to give them. One idea I had is letting you see the damage dice of monsters, since, y'know, dice.

**Comment by sfsworms** [comment]
Again, it doesn't really make sense and doesn't seem to fit in the core game. Likewise, I'm not sure it make sense to give melee or archery to a table top RPG player.

**Comment by DracoGriffin** [comment]
I think as it is, would be better suited for the Crazy Cataclysm mod. Otherwise, the unique trait is just a bit ... flimsy. It is interesting but as others point out, it feels out of place. Maybe something with reading slightly faster, like some sort of [speedreading technique](http://www.learningtechniques.com/speed-reading-techniques.html) `After constantly referring to your manual, you've managed to pick out important details from printed materials much faster than others`.

Or some sort of 4th wall ability (which can still work, go with me here), similar to the [Zombie Survival Guide book](https://github.com/CleverRaven/Cataclysm-DDA/blob/d44e02a884d87bcef8ca02f80e7c8b3a919c9899/data/json/items/book/misc.json#L5), so maybe the trait is they can understand more about hidden abilities of monsters through "experienced guessing"? `Hmm, that monster reminds me of the minions of Dr. Acula, they were capable of [doing ability]!`

**Comment by Xhuis** [comment]
I think what I'll do is -
* Remove the ability to see exact HP from the Game Master
* Give the Game Master a trait for easier reading as @DracoGriffin suggested
* Hold onto the idea for monster HP, and save it for either a bionic or item (or both!)

**Comment by DracoGriffin** [comment]
>Give the Game Master a trait for easier reading as @DracoGriffin suggested

I should probably mention, I'd hazard that it should stack with the Faster Reading trait rather than acting as a profession duplicate. Something like Faster Reading is a speedreading technique, and the Game Master trait is, through experience and constantly referencing sources (sound familiar anyone on this repo), is capable of understanding technical terms and such, and so is able to skim over the nuances that others may have to read in order to understand. Good enough rationale for me anyway.

**Comment by Xhuis** [comment]
I changed the trait: now, it increases reading speed by 10% (half as effective as Fast Reader), in addition to a unique effect - removing the reading speed bonus for books too complex for you to understand. So long as you have the trait, you can understand any book you have the skill requirement to read.

**Comment by nexusmrsep** [comment]
Bionic Health Analizer CBM sounds legit too. With a lore snippet that its dedicated for medical personell as a built in tool for quick diagnosis application.

**Comment by Mecares** [comment]
Health Analysis does not sound like it would work well on Robots or walking dead, maybe something like structural damage analysator instead?

**Comment by Xhuis** [comment]
This should be ready to go.

File: data/json/items/book/speech.json [hunk 1]
```diff
@@ -52,5 +52,24 @@
     "max_level": 6,
     "intelligence": 8,
     "time": 30
+  },
+  {
+    "id": "dnd_handbook",
```
Comment (Rivet-the-Zombie) [inline data/json/items/book/speech.json]: Surely this should spawn in-world too.

Comment (Xhuis) [reply data/json/items/book/speech.json]: I was initially against it, as it doesn't train skills that existing textbooks don't do better already, and intended it as a profession book ala Hitchhiker's Guide to the Cataclysm. If you want, I can put it into the loot tables.

Comment (Rivet-the-Zombie) [reply data/json/items/book/speech.json]: My inner nerd would really like that. Perhaps as a rare spawn in houses and maybe coffee shops?

Comment (Xhuis) [reply data/json/items/book/speech.json]: I might sprinkle it into the recently-added LAN centers as well, I feel like it'd fit there.

**Comment by Xhuis** [comment]
Sprinkled it into a few item groups, including zombie children, school libraries, lab dorms (what, you think a lab full of nerds wouldn't have it?) and other places as a rare spawn.

**Comment by Xhuis** [comment]
What the hell is your problem, git? Why can't you work?

**Comment by Xhuis** [comment]
Commits squashed, issues fixed. Only gave me a few aneurysms. This should be ready to go.

**Pull request merged**